src/Subscription/Engine/ErrorDetected.php

```diff
@@ -16,15 +15,15 @@ final class ErrorDetected extends RuntimeException
     public function __construct(
         public readonly array $errors,
     ) {
-        $sentences = array_map(
-            static fn (Error $error) => sprintf(
-                'Subscription %s: %s',
-                $error->subscriptionId,
-                $error->message,
+        parent::__construct(
+            sprintf(
+                '%s error(s) in subscription engine detected. First error in "%s" subscription: %s',
+                count($errors),
+                $errors[0]->subscriptionId,
+                $errors[0]->message,
             ),
-            $errors,
+            0,
+            $errors[0]->throwable,
```

Comment:
maybe use here a named parameter to mitigate the error code? Or add this to a test 🙈